docs/customization.rst

```diff
@@ -145,6 +145,23 @@ With this hook you can add additional test cases that will be executed in Hypoth
             Case(endpoint=context.endpoint, query={"foo": "bar"})
         )
 
+``add_case``
+~~~~~~~~~~~~~~~~~~~~~~~
+
+For each add_case hook and for each endpoint, we create an additional, duplicate test case. We pass the Case object from the duplicate test to the add_case hook.
+The user may change the Case object (and therefore the request data) before the request is sent to the server. The add_case allows the user to target specific
+behavior in the API by changing specific details of the duplicate request.
+
+.. code:: python
+
+    def add_case(
```

Comment:
Even though we have access to `endpoint` via case I think that the first argument should be hook context (at least for the forward-compatibility). What do you think?

Also, the function signature will be quite short - probably it can fit a single line